.github/workflows/v2dl3-vegas.yml

```diff
@@ -0,0 +1,81 @@
+---
+name: v2dl3-vegas-CI
+
+# Controls when the workflow will run
+on:
+  pull_request :
+    paths-ignore:
+      - 'tests/**'
+      - 'utils/v2dl3-vegas-docker/**'
+      - 'utils/download_ED_test_data.sh'
+      - '.github/v2dl3ED.yml'
+      - '.git*'
+      - 'CITATION.cff'
+      - 'environment*.yml'
+      - 'LICENSE'
+      - 'README.md'
+
+jobs:
+  v2dl3-vegas:
+    # The type of runner that the job will run on
+    runs-on: ubuntu-latest
+    # Environment vars for the runner
+    env:
+      DOCKER_USERNAME: ${{ secrets.DOCKER_USERNAME }}
+      DOCKER_TOKEN: ${{ secrets.DOCKER_TOKEN }}
+      DOCKER_IMAGE: veritasacc/vegas:v2dl3-CI
+      DOCKER_WORKSPACE: V2DL3-CI
+      # New 'test' outputs
+      NEW_DIR: ./new-fits
+      # 'Control' outputs from main branch
+      CONTROL_DIR: ./control-fits
+
+    # Steps represent a sequence of tasks that will be executed as part of the job
+    steps:
+      # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
+      - name: Checkout new branch
+        uses: actions/checkout@v3
+
+      - name: Pull docker image
+        run: |
+          docker login --username=$DOCKER_USERNAME --password=$DOCKER_TOKEN
+          docker pull $DOCKER_IMAGE
+
+      # Runs "utils/vegas_docker_test.runs.sh" within the docker image.
+      # First copies out the test scripts from the new branch so that they aren't overwritten by the main branch pull
+      - name: Install new branch and generate .fits outputs
+        run: |
+          cp utils/vegas_docker_test_runs.sh ../vegas_docker_test_runs_static.sh
+          cp utils/compare_fits_dirs.sh ../compare_fits_dirs.sh
+          docker run -p 80:80 -v $GITHUB_WORKSPACE:/$DOCKER_WORKSPACE $DOCKER_IMAGE /bin/bash -c "\
+          cd $DOCKER_WORKSPACE \
+          && bash utils/vegas_docker_test_runs.sh OUTDIR=$NEW_DIR \
+          "
+          cp -r $NEW_DIR ../$NEW_DIR
+          sudo rm -rf $GITHUB_WORKSPACE/*
+
+      - name: Checkout main branch
+        uses: actions/checkout@v3
+        with:
+          ref: main
+
+      - name: Install main branch and generate control outputs
+        run: |
+          cp ../vegas_docker_test_runs_static.sh utils/vegas_docker_test_runs_static.sh
+          docker run -p 80:80 -v $GITHUB_WORKSPACE:/$DOCKER_WORKSPACE $DOCKER_IMAGE /bin/bash -c "\
+          cd $DOCKER_WORKSPACE \
+          && bash utils/vegas_docker_test_runs_static.sh OUTDIR=$CONTROL_DIR \
+          "
+          cp -r $CONTROL_DIR ../$CONTROL_DIR
+
+      - name: Install fitsdiff
```

Comment:
Why fitsdiff and not using the script already there? https://github.com/VERITAS-Observatory/V2DL3/blob/main/pyV2DL3/script/compareFitsFiles.py should do the job, or?

Reply:
Mine works recursively for all fits in any subdir of the two given directories. Fits files of the same child paths will be compared. 

e.g `new_fits/test_dirs/fits_files` compares all the `fits_files` in `test_dirs` to their matching paths under `control_fits/test_dirs/fits_files`

Then only ones with any differences are printed to console. If at least one difference is detected in any file, the test "fails" and you can review just the differences.

It is done this way to dynamically handle if the paths or number of tests or inputs change in the future. In this way, it's just given the top level directory of the test outputs from the incoming branch and the main branch.